Docs/ReleaseNotes_Swift5.md

```diff
@@ -156,3 +156,10 @@ A few notes:
 * The exception-throwing API will be deprecated in a future release. It is now marked as `@available(…, deprecated: 100000, …)`, which matches what you would see if `API_TO_BE_DEPRECATED` was used in a Objective-C header.
 
 * Subclassing `NSFileHandle` is strongly discouraged. Many of the new methods are `public` and cannot be overridden.
+
+## `NSString(bytesNoCopy:length:encoding:freeWhenDone:)` deviations from reference implementation
+
+On Windows, `NSString(bytesNoCopy:length:encoding:freeWhenDone:)` deviates from
+the behaviour on Darwin and uses the buffer's `deallocate` routine rather than
+`free`.  This is done to ensure that small buffers which are stack allocated
+through `alloca` can be properly released.
```

Comment:
This is actually to ensure that buffers that are allocated with `UnsafeMutableRawPointer` are correctly released. If `UnsafeMutableRawPointer` may be using `alloca` under the covers, we can note that, but the contract isn't specific to `alloca` (and indeed someone who used `alloca` on their own is expected to `freeWhenDone: false`).

Reply:
… is `UM{R,}P` using `alloca`?

Reply:
Because I'm not sure how it squares with the contract of its methods.

Reply:
(This is relevant because we want to be able to invoke `.deallocate()` on things the user can give us guarantees on, and if someone rebases a pointer it may not know not to free that pointer because it's on the stack.)

Reply:
IIRC, Windows has issues with `alloca`, `posix_memalign` (spelt `_aligned_alloc`), and `malloc`.  The first is largely a bad idea (though clang will sometimes emulate it).  `_aligned_alloc` must use `_aligned_free` and `malloc` must use `free`.  Other platforms are not as strict. I would need to go digging into the stdlib implementation, but IIRC, Andy had some trickery for small allocations.